### PR TITLE
Unetrpp stem settings

### PR DESCRIPTION
**Pull request by colon3ltocard**
* Makes the first downsampling of UNETR++ a tunable parameter
* Improve doc
* Fixes single node multi-GPU training @flyIchtus

### CODE REVIEW

**Review by flyIchtus — COMMENTED** [review]
Why removing the two last .cpu() commands ? I added them otherwise there was an OOM in large Titan test runs.

File: py4cast/models/vision/unetrpp.py [hunk 1]
```diff
@@ -304,8 +304,7 @@ def __init__(
         in_channels=4,
         dropout=0.0,
         transformer_dropout_rate=0.1,
-        subsampling_rate: int = 4,
-        **kwargs,
+        downsampling_rate: int = 4,
```
Comment (flyIchtus) [inline py4cast/models/vision/unetrpp.py:307]: int=4 as default seems quite aggressive (mere remark)

**Review by flyIchtus — APPROVED** [review]

**Review by flyIchtus — APPROVED** [review]

File: py4cast/models/vision/unetrpp.py (hunk 1 above)
Comment (colon3ltocard) [reply py4cast/models/vision/unetrpp.py:307]: It is the default value of the reference implementation. I tend to stick to those as defaults. Maybe if we identify a good candidate we will switch to its settings ?

**Comment by colon3ltocard** [comment]
> Why removing the two last .cpu() commands ? I added them otherwise there was an OOM in large Titan test runs.

Because it breaks multi- GPU training ! I did the .cpu() after gathering though what do you think ?

File: py4cast/models/vision/unetrpp.py (hunk 1 above)
Comment (flyIchtus) [reply py4cast/models/vision/unetrpp.py:307]: let's keep it this way, my remark was no realy to the point

**Comment by colon3ltocard** [comment]
> > > Why removing the two last .cpu() commands ? I added them otherwise there was an OOM in large Titan test runs.
> > 
> > 
> > Because it breaks multi- GPU training ! I did the .cpu() after gathering though what do you think ?
> 
> ow, I did not pay attention ! however, not sure it will work. The pb, as far as I understand, is that the .cpu() at gather happens "too late", ie at the end of an epoch. The list of losses is growing continuously with data on gpu during the whole epoch (which does not seem very useful). A question : why does it break multi-gpu training ? Couldn't we add a torch.no_grad context before computing the metrics in the observer (would fix any gradient-related error) ?

I propose we solve this in a dedicated issue.

**Review by cbovalo — APPROVED** [review]

**Review by LBerth — APPROVED** [review]
